app/models/users/project_role_cache.rb

```diff
@@ -40,8 +40,9 @@ def fetch(project)
   private
 
   def roles(project)
-    # No role on archived projects
-    return [] unless !project || project&.active?
+    # Project is nil if checking global role
+    # No role on archived projects, unless the activity is being changed
```

Comment:
I'd rather call it "active state" then "activity" since the later is already used in different contexts in OP.

Reply:
good point :+1:

Reply:
Done